### ISSUE TITLE
Data received but not shown in UI

### ISSUE THREAD

**Issue by PioneerX** [tananaev/traccar#2409]
I have a fresh install of traccar on a RaspberryPi3 and all seemed well with the install. However when I test the tracking from my iPhone (traccar client app) I get the data incoming but it does not display on the UI.

I installed the iPhone app and interfaced it with your demo site @ demo.traccar.org and all works well.....

http://imgur.com/uAruNvO

...however when I swap out the 'demo.traccar.org' URL for my own I can see the 'Last Update' date change but when you click the device no 'Attributes' are displayed.

http://imgur.com/7FfAimQ

I have checked the database behind my instance of traccar and the position records are shown....

http://imgur.com/YjvcSK3

...this was true for both the inbuilt database and the MySQL I'm now using. I changed out the database so I could see if traccar was infact storing the data. The incoming device data also shows up in the tracker-server.log file.....
```
2016-10-05 16:25:05  INFO: [EBEE524A] connected
2016-10-05 16:25:06 DEBUG: [EBEE524A: 5055 < 31.103.190.219] HEX: 474554202f3f69643d3431353234372674696d657374616d703d31343735363831313031266c61743d35322e343735383334266c6f6e3d2d302e3931333837372673706565643d302662656172696e673d3026616c$
2016-10-05 16:25:06 DEBUG: [EBEE524A: 5055 > 31.103.190.219] HEX: 485454502f312e3120323030204f4b0d0a436f6e74656e742d4c656e6774683a20300d0a0d0a
2016-10-05 16:25:07  INFO: [EBEE524A] id: 415247, time: 2016-10-05 16:25:01, lat: 52.47583, lon: -0.91388, speed: 0.0, course: 0.0
2016-10-05 16:25:08  INFO: [EBEE524A] disconnected
2016-10-05 16:28:04  INFO: [5557980A] connected
2016-10-05 16:28:04 DEBUG: [5557980A: 5055 < 31.103.190.219] HEX: 474554202f3f69643d3431353234372674696d657374616d703d31343735363831323739266c61743d35322e343735383237266c6f6e3d2d302e3931333934372673706565643d302662656172696e673d3026616c$
2016-10-05 16:28:04 DEBUG: [5557980A: 5055 > 31.103.190.219] HEX: 485454502f312e3120323030204f4b0d0a436f6e74656e742d4c656e6774683a20300d0a0d0a
2016-10-05 16:28:04  INFO: [5557980A] id: 415247, time: 2016-10-05 16:27:59, lat: 52.47583, lon: -0.91395, speed: 0.0, course: 0.0
2016-10-05 16:28:05  INFO: [5557980A] disconnected
2016-10-05 16:29:00  INFO: [BBD9971C] connected
2016-10-05 16:29:00 DEBUG: [BBD9971C: 5055 < 31.103.190.219] HEX: 474554202f3f69643d3431353234372674696d657374616d703d31343735363831333339266c61743d35322e343735383338266c6f6e3d2d302e3931343133382673706565643d302662656172696e673d3026616c$
2016-10-05 16:29:00 DEBUG: [BBD9971C: 5055 > 31.103.190.219] HEX: 485454502f312e3120323030204f4b0d0a436f6e74656e742d4c656e6774683a20300d0a0d0a
2016-10-05 16:29:00  INFO: [BBD9971C] id: 415247, time: 2016-10-05 16:28:59, lat: 52.47584, lon: -0.91414, speed: 0.0, course: 0.0
2016-10-05 16:29:02  INFO: [BBD9971C] disconnected
```
If you have any ideas or ca help me diagnose this issue I would be grateful.

System Details:
Physical Device = Raspberry Pi 3B (Quad Core ARM, 1Gb RAM, 8Gb SD)
OS = minibian (stripped down version of Rasbian from https://minibianpi.wordpress.com/) as used on all my servers
JAVA = Java(TM) SE Runtime Environment (build 1.8.0_65-b17)

Traccar Details (from log):
```
2016-10-05 14:26:22  INFO: Operating system name: Linux version: 4.1.18-v7+ architecture: arm
2016-10-05 14:26:22  INFO: Java runtime name: Java HotSpot(TM) Client VM vendor: Oracle Corporation version: 25.65-b01
2016-10-05 14:26:22  INFO: Memory limit heap: 235mb non-heap: 0mb
2016-10-05 14:26:22  INFO: Character encoding: UTF-8 charset: UTF-8
2016-10-05 14:26:22  INFO: Version: 3.7-SNAPSHOT
```

**Comment by tananaev** [comment]
Everything seems to be working fine. Try to refresh the page, you should be able to see the data.

**Comment by PioneerX** [comment]
Refreshing the page has no effect on the display of the data. I agree that everything seems to be working fine but we still dont get any data on the UI despite it being held in the logs and DB.

I did some more digging in the logs and while the tracker-server.log is all normal the wrapper.log.20161005 contains a 32bit warning message for the wrapper.

STATUS | wrapper  | 2016/10/05 15:51:19 | <-- Wrapper Stopped
STATUS | wrapper  | 2016/10/05 16:12:09 | --> Wrapper Started as Daemon
STATUS | wrapper  | 2016/10/05 16:12:09 | Java Service Wrapper Community Edition 32-bit 3.5.28
STATUS | wrapper  | 2016/10/05 16:12:09 |   Copyright (C) 1999-2016 Tanuki Software, Ltd. All Rights Reserved.
STATUS | wrapper  | 2016/10/05 16:12:09 |     http://wrapper.tanukisoftware.com
STATUS | wrapper  | 2016/10/05 16:12:09 |
STATUS | wrapper  | 2016/10/05 16:12:09 | Launching a JVM...
INFO   | jvm 1    | 2016/10/05 16:12:09 | WrapperManager: Initializing...
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager: WARNING - Unable to load the Wrapper's native library 'libwrapper.so'.
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           The file is located on the path at the following location but
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           could not be loaded:
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:             /opt/traccar/bin/../lib/libwrapper.so
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           Please verify that the file is both readable and executable by the
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           current user and that the file has not been corrupted in any way.
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           One common cause of this problem is running a 32-bit version
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           of the Wrapper with a 64-bit version of Java, or vica versa.
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           This is a 32-bit JVM.
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           Reported cause:
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:             /opt/traccar/lib/libwrapper.so: /opt/traccar/lib/libwrapper.so: cannot open shared object file: No such file or directory
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:           System signals will not be handled correctly.
INFO   | jvm 1    | 2016/10/05 16:12:10 | WrapperManager:


....since the RaspberryPi3 is 64bit ARMv8 I'm wondering if the miss-match is related despite the application starting up and running seemingly as normal.

**Comment by PioneerX** [comment]
After some more digging I found that if I access the server directly everything works but accessing through my reverse proxy loads everything but the data. This points the issue at my apache reverse proxy rather than Traccar but for the sake of others I will detail the situation and fix below.

My Traccar is running on a Raspberry Pi 3 (64bit) and is accessed via an apache reverse proxy. Using the guide @ https://www.traccar.org/secure-connection/ I discovered that Traccar requires the proxy to pass both http AND socket connections.....I was unaware of the later so my proxy was not setup for sockets.

After adding the proxy_wstunnel module to my apache (I had to back port it from 2.2.24 to 2.2.15 because my ReverseProxy is on an outdated OpenVZ VM) and adding the ProxyPass lines for both http AND ws the problem was solved.

Resources used to solve the problem:
- https://www.traccar.org/secure-connection/ for apache config information on HTTP and WS
- https://www.amoss.me.uk/2013/06/apache-2-2-websocket-proxying-ubuntu-mod_proxy_wstunnel/ to backport the proxy_wstunnel module to my old apache version

I hope this helps others
